lib/algo/manifold_interp/SnapshotInterpolator.cpp

```diff
@@ -0,0 +1,221 @@
+/******************************************************************************
+ *
+ * Copyright (c) 2013-2024, Lawrence Livermore National Security, LLC
+ * and other libROM project developers. See the top-level COPYRIGHT
+ * file for details.
+ *
+ * SPDX-License-Identifier: (Apache-2.0 OR MIT)
+ *
+ *****************************************************************************/
+
+/**
+ * Implements PCHIP algorithm.  Based on "A METHOD FOR CONSTRUCTING LOCAL MONOTONE
+ * PIECEWISE CUBIC INTERPOLANTS", Fritchs and Butland (1984).  as well as "MONOTONE
+ * PIECEWISE CUBIC INTERPOLATION," Fritsch and Carlson (1980)
+ *
+ */
+#include "SnapshotInterpolator.h"
+
+#include <limits.h>
+#include <cmath>
+#include "linalg/Vector.h"
+#include "mpi.h"
+
+/* Use C++11 built-in shared pointers if available; else fallback to Boost. */
+#if __cplusplus >= 201103L
+#include <memory>
+#else
+#include <boost/shared_ptr.hpp>
+#endif
+using namespace std;
+
+namespace CAROM {
+
+SnapshotInterpolator::SnapshotInterpolator()
+{
+
+}
+
+
+std::vector<Vector*> SnapshotInterpolator::interpolate(std::vector<Vector*>
+        snapshot_ts,
+        std::vector<Vector*> snapshots,
+        std::vector<Vector*> output_ts)
+{
+    CAROM_VERIFY(snapshot_ts.size() == snapshots.size());
+    CAROM_VERIFY(snapshot_ts.size() > 0);
+    CAROM_VERIFY(output_ts.size() > 0);
+
+    int n_out = output_ts.size();
+    int n_snap = snapshots.size();
+    int n_dim = snapshots[0]->dim();
+
+    std::vector<Vector *> output_snapshots;
+    for(int i = 0; i < n_out; ++i)
+    {
+        Vector* temp_snapshot =new Vector(snapshots[0]->dim(),
```

Comment:
Instead of using pointers and having to keep track of pointer ownership, it may be simpler to use `std::vector<Vector>` and `emplace_back` for each `i`.

Reply:
Unfortunately, this appears to be inconsistent with choices made in the DMD class.  Keeping the current implementation for now.